services/kvstore/src/main/java/tech/pegasys/pantheon/services/kvstore/RocksDbKeyValueStorage.java

```diff
@@ -44,19 +48,45 @@
   private final TransactionDB db;
   private final AtomicBoolean closed = new AtomicBoolean(false);
 
+  private final OperationTimer readLatency;
+  private final OperationTimer removeLatency;
+  private final OperationTimer writeLatency;
+  private final OperationTimer commitLatency;
+  private final Counter rollbackCount;
+
   static {
     RocksDB.loadLibrary();
   }
 
-  public static KeyValueStorage create(final Path storageDirectory) throws StorageException {
-    return new RocksDbKeyValueStorage(storageDirectory);
+  public static KeyValueStorage create(
+      final Path storageDirectory, final MetricsSystem metricsSystem) throws StorageException {
+    return new RocksDbKeyValueStorage(storageDirectory, metricsSystem);
   }
 
-  private RocksDbKeyValueStorage(final Path storageDirectory) {
+  private RocksDbKeyValueStorage(final Path storageDirectory, final MetricsSystem metricsSystem) {
     try {
       options = new Options().setCreateIfMissing(true);
       txOptions = new TransactionDBOptions();
       db = TransactionDB.open(options, txOptions, storageDirectory.toString());
+
+      readLatency =
+          metricsSystem.createTimer(
+              MetricCategory.ROCKSDB, "read_latency_seconds", "Latency for read from RocksDB.");
+      removeLatency =
+          metricsSystem.createTimer(
+              MetricCategory.ROCKSDB,
+              "remove_latency_seconds",
+              "Latency of remove requests from RocksDB.");
+      writeLatency =
+          metricsSystem.createTimer(
+              MetricCategory.ROCKSDB, "write_latency_seconds", "Latency for write to RocksDB.");
+      commitLatency =
+          metricsSystem.createTimer(
+              MetricCategory.ROCKSDB, "commit_latency_seconds", "Latency for write to RocksDB.");
+
+      rollbackCount =
+          metricsSystem.createCounter(
+              MetricCategory.ROCKSDB, "rollback_count", "Number of transactions rolled back.");
```

Comment:
Can we make the description "Number of RocksDB transactions rolled back."?  I know it will have rocksdb in the key as well but it would be easy to confuse this with number of EVM transactions rolled back.

Reply:
Done